harness/tests/experiment/pytorch/test_deepspeed_autotuning.py

```diff
@@ -134,75 +170,67 @@ def _process_operations(self, operations: List[searcher.Operation]) -> None:
-                requestId=str(op.request_id),
-                metric=metric,
-                validateAfterLength=str(op.length),
-            )
             self.metric_index += 1
-            self.events_count += 1
-            event = bindings.v1SearcherEvent(
-                id=self.events_count, validationCompleted=validation_completed
-            )
-            self.events_queue.append(event)
-
-            trial_progress = bindings.v1TrialProgress(
-                requestId=str(op.request_id), partialUnits=float(op.length)
-            )
-            self.events_count += 1
-            event = bindings.v1SearcherEvent(id=self.events_count, trialProgress=trial_progress)
-            self.events_queue.append(event)
-
-        if isinstance(op, searcher.Create):
+            if ERROR_METRIC_NAME in metric:
+                trial_exited_early = bindings.v1TrialExitedEarly(
+                    requestId=str(op.request_id),
+                    exitedReason=bindings.v1TrialExitedEarlyExitedReason.EXITED_REASON_UNSPECIFIED,
+                )
+                self.events_count += 1
+                event = bindings.v1SearcherEvent(
+                    id=self.events_count, trialExitedEarly=trial_exited_early
+                )
+                self.events_queue.append(event)
+
+                trial_closed = bindings.v1TrialClosed(requestId=str(op.request_id))
+                self.events_count += 1
+                event = bindings.v1SearcherEvent(id=self.events_count, trialClosed=trial_closed)
+                self.events_queue.append(event)
+            else:
+                validation_completed = bindings.v1ValidationCompleted(
+                    requestId=str(op.request_id),
+                    metric=metric,
+                    validateAfterLength=str(op.length),
+                )
+
+                self.events_count += 1
+                event = bindings.v1SearcherEvent(
+                    id=self.events_count, validationCompleted=validation_completed
+                )
+                self.events_queue.append(event)
+
+                # Send 1.0 to signal it was completed
+                trial_progress = bindings.v1TrialProgress(
+                    requestId=str(op.request_id), partialUnits=1.0
+                )
+                self.events_count += 1
+                event = bindings.v1SearcherEvent(id=self.events_count, trialProgress=trial_progress)
+                self.events_queue.append(event)
+
+        elif isinstance(op, searcher.Create):
             trial_created = bindings.v1TrialCreated(requestId=str(op.request_id))
             self.events_count += 1
             event = bindings.v1SearcherEvent(id=self.events_count, trialCreated=trial_created)
             self.events_queue.append(event)
 
-        if isinstance(op, searcher.Progress):  # no events
-            self.overall_progress
+        elif isinstance(op, searcher.Progress):  # no events
+            pass
```

Comment:
Why not do something with the value returned by the `Progress` operation?

Reply:
I could look into doing that